guardrails-scripts/yb-voyager-pg-grant-migration-permissions.sql

```diff
@@ -1,83 +1,86 @@
+--- psql -h <host> -d <database> -U <username> -v voyager_user='ybvoyager' -v schema_list='schema1,public,schema2' -v is_live_migration=1 -v is_live_migration_fall_back=0 -v replication_group='replication_group' -v original_owner_of_tables='postgres' -f /home/ubuntu/yb-voyager-pg-grant-migration-permissions.sql
 --- Parameters:
 --- <host>: The hostname of the PostgreSQL server.
 --- <database>: The name of the database to connect to.
 --- <username>: The username to connect with.
 --- <voyager_user>: The database user for which permissions are being granted.
 --- <schema_list>: A comma-separated list of schemas to grant permissions on. Example 'schema1,public,schema2'.
+--- <is_live_migration>: A flag indicating if this is a live migration (1 for true, 0 for false). If set to 0 then the script will check for permissions for an offline migration.
+--- <is_live_migration_fall_back>: A flag indicating if this is a live migration with fallback (1 for true, 0 for false). If set to 0 then the script will detect permissions for live migration with fall-forward. Should only be set to 1 when is_live_migration is also set to 1. Does not need to be provided unless is_live_migration is set to 1.
 --- <replication_group>: The name of the replication group to be created. Not needed for offline migration.
 --- <original_owner_of_tables>: The original owner of the tables to be added to the replication group. Not needed for offline migration.
---- <is_live_migration>: A flag indicating if this is a live migration (1 for true, 0 for false). If set to 0 then the script will check for permissions for an offline migration.
---- <is_live_migration_fall_back>: A flag indicating if this is a live migration with fallback (1 for true, 0 for false). If set to 0 then the script will detect permissions for live migration with fall-forward. Should only be set to 1 when is_live_migration is also set to 1.
+
 \echo ''
 \echo '--- Checking Variables ---'
 
 -- Check if voyager_user is provided
 \if :{?voyager_user}
-    \echo 'Database user (voyager_user) is provided: ':voyager_user
+    \echo 'Voyager user is provided: ':voyager_user
 \else
-    \echo 'Error: Database user (voyager_user) is not provided!'
+    \echo 'Error: voyager_user flag is not provided!'
     \q
 \endif
 
 -- Check if schema_list is provided
 \if :{?schema_list}
-    \echo 'Schema list (schema_list) is provided: ':schema_list
+    \echo 'Schema list is provided: ':schema_list
 \else
-    \echo 'Error: Schema list (schema_list) is not provided!'
+    \echo 'Error: schema_list flag is not provided!'
     \q
 \endif
 
 -- Check if is_live_migration is provided
 \if :{?is_live_migration}
-    \echo 'Live migration flag (is_live_migration) is provided: ':is_live_migration
+    \echo 'Live migration flag is provided: ':is_live_migration
 \else
-    \echo 'Error: Live migration flag (is_live_migration) is not provided!'
+    \echo 'Error: is_live_migration flag is not provided!'
```

Comment:
Why is this flag mandatory?

Reply:
Hmm, I still think it need not be mandatory. It's fine if a lot of checks use it, the default value can be 0, no? Can be discussed separately since you're not really making the change as part of this PR.

Reply:
Hmm yeah I can pick this up when I am working on the original owner of tables issue